assets/dev/js/editor/utils/container-helper.js

```diff
@@ -70,125 +70,129 @@ export class ContainerHelper {
-					},
-				};
-
-				if ( createForTarget ) {
-					$e.run( 'document/elements/settings', { container, settings } );
-					newContainer = container;
-				} else {
-					newContainer = ContainerHelper.createContainer( settings, container, options );
-				}
-
-				settings = {
-					width: {
-						unit: '%',
-						size: '50',
-					},
-					width_mobile: {
-						unit: '%',
-						size: '100',
-					},
-				};
-
-				ContainerHelper.createContainer( settings, newContainer, { edit: false } );
-
-				const rightContainer = ContainerHelper.createContainer( {
-					...settings,
-					padding: { size: '' }, // Create the right Container with 0 padding (default is 10px) to fix UI (ED-4900).
-					flex_gap: {
-						unit: 'px',
-						size: 0, // Set the gap to 0 to override the default inherited from `Site Settings`.
-					},
-				}, newContainer, { edit: false } );
-
-				ContainerHelper.createContainers( 2, {}, rightContainer, { edit: false } );
-
-				break;
-
-			// Containers by preset.
-			default:
-				const sizes = preset.split( '-' ),
-					// Map rounded, user-readable sizes to actual percentages.
-					sizesMap = {
-						33: '33.3333',
-						66: '66.6666',
+			} ),
+			{ createWrapper = false } = options;
+
+		try {
+			switch ( preset ) {
+				// Single Container without sub Containers.
+				case '100':
+					newContainer = ContainerHelper.createContainer( {}, target, options );
+					break;
+
+				// Exceptional preset.
+				case 'c100-c50-50':
+					settings = {
+						flex_direction: ContainerHelper.DIRECTION_ROW,
+						flex_wrap: 'wrap',
+						flex_gap: {
+							unit: 'px',
+							size: 0, // Set the gap to 0 to override the default inherited from `Site Settings`.
+						},
 					};
 
-				settings = {
-					flex_direction: this.DIRECTION_ROW,
-					flex_wrap: 'wrap',
-					flex_gap: {
-						unit: 'px',
-						size: 0, // Set the gap to 0 to override the default inherited from `Site Settings`.
-					},
-				};
-
-				// Create a parent container to contain all of the sub containers.
-				let parentContainer;
-
-				if ( createForTarget ) {
-					$e.run( 'document/elements/settings', {
-						container,
-						settings,
-					} );
-
-					parentContainer = container;
-				} else {
-					parentContainer = this.createContainer( settings, container, options );
-				}
-
-				// Create all sub containers using the sizes array.
-				// Use flex basis to make the sizes explicit.
-				sizes.forEach( ( size ) => {
-					size = sizesMap[ size ] || size;
+					if ( createWrapper ) {
+						$e.run( 'document/elements/settings', { container: target, settings } );
+						newContainer = target;
+					} else {
+						newContainer = ContainerHelper.createContainer( settings, target, options );
+					}
 
-					this.createContainer( {
-						flex_direction: this.DIRECTION_COLUMN,
+					settings = {
 						width: {
 							unit: '%',
-							size,
+							size: '50',
```

Comment:
Check with @StyleShit

Reply:
@StyleShit 
we need somehow add that logic to your branch, and I think it should be committed by you.
also, if you can think of a way to minimize the method/part it, please suggest.